relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -479,7 +479,23 @@ fn extract_span_metrics(
                         Some((_method, url)) => {
                             let url = SchemeDomainPort::from(url);
                             match (url.domain, url.port) {
-                                (Some(domain), Some(port)) => Some(format!("{}:{}", domain, port)),
+                                (Some(domain), Some(port)) => {
+                                    let tokens = domain.split('.').collect::<Vec<&str>>();
+                                    if tokens.len() > 2 {
+                                        // sth.subdomain.domain.tld => *.domain.tld
+                                        // TODO(iker): find a proper way to do
+                                        // this and consider cloud domains, e.g.
+                                        // <name>.service.<cloud-region>.<whatever>
+                                        let extracted_domain: Vec<&&str> =
+                                            tokens.iter().rev().take(2).rev().collect();
+                                        Some(format!(
+                                            "*.{}.{}:{}",
+                                            extracted_domain[0], extracted_domain[1], port
+                                        ))
+                                    } else {
+                                        Some(format!("{}:{}", domain, port))
+                                    }
```

Comment:
nit: Should be possible to do this without allocating a vector (and without the `else` branch), but I don't have the code for it ready so feel free to ignore.